### PR TITLE
[297] Move preview templates to app/views folder

### LINKED ISSUE #297: Refactor the preview templates to be located in `app/views` folder
### Feature request
Following the discussion started in https://github.com/github/view_component/issues/289#issuecomment-612612056 we should make use of the `app/views` folder for our preview templates. 

Also we might not need to namespace the previews controller with `Rails`.

### Motivation
Just to improve code quality and maintainability in the gem

### PR DESCRIPTION
**Pull request by juanmanuelramallo**
Resolves #297 

### Summary

- Removed unused rails module in railties folder
- Removed rails module from the view_components_controller
- Moved index, preview and previews templates to app/views/view_components

### CODE REVIEW

File: lib/view_component/engine.rb [hunk 1]
```diff
@@ -64,8 +64,8 @@ class Engine < Rails::Engine # :nodoc:
 
       if options.show_previews
         app.routes.prepend do
-          get "/rails/view_components"       => "rails/view_components#index", :internal => true
-          get "/rails/view_components/*path" => "rails/view_components#previews", :internal => true
+          get "/rails/view_components"       => "view_components#index", :internal => true
+          get "/rails/view_components/*path" => "view_components#previews", :internal => true
```
Comment (jonspalmer) [inline lib/view_component/engine.rb:68]: I've often thought that we should make the mount point configurable but that is a separate PR.

File: app/controllers/view_components_controller.rb [hunk 2]
```diff
@@ -16,26 +16,19 @@ class Rails::ViewComponentsController < Rails::ApplicationController # :nodoc:
   def index
     @previews = ViewComponent::Preview.all
     @page_title = "Component Previews"
-    # rubocop:disable GitHub/RailsControllerRenderPathsExist
-    render "components/index"
-    # rubocop:enable GitHub/RailsControllerRenderPathsExist
```
Comment (jonspalmer) [inline app/controllers/view_components_controller.rb:21]: getting rid of these and the rubocop stuff is a great sign we 'did it right'

**Comment by jonspalmer** [comment]
List looks awesome 💯

File: lib/view_component/engine.rb (hunk 1 above)
Comment (joelhawksley) [reply lib/view_component/engine.rb:68]: @juanmanuelramallo would you be willing to file this as an issue?

Comment (juanmanuelramallo) [reply lib/view_component/engine.rb:68]: https://github.com/github/view_component/issues/326

How does it look?

Comment (joelhawksley) [reply lib/view_component/engine.rb:68]: @juanmanuelramallo looks good, thanks!

**Pull request merged**